### DISCUSSION TITLE
Automating sanity testing for flux-core

### DISCUSSION THREAD

**Discussion by wspear** [flux-framework/flux-core#6177] [Q&A] [answered]
I'm working on the e4s testsuite which runs quick sanity-checks on spack-installed packages provided by e4s (https://e4s-project.github.io/). flux-core is one of those packages. Is there a good way to run a totally scripted test of a pre-installed flux-core? 

If I were testing manually I would do `flux start`, run a couple flux commands, maybe launch a trivial mpi application and exit but since `flux start` enters a new console session I can't find a way to script what happens after that. Redirecting commands to `flux start` via stdin doesn't seem to work.

Flux can not be running in the test environment prior to executing the test script. Any ideas how I can proceed here?  Thanks!

**Comment by grondo** [comment, marked as answer]
If you run `flux start COMMAND` then the Flux test session will run `COMMAND` and exit with the exit status of the command.
So you could use a script for COMMAND and run a few sanity commands, exiting with error on failure.

It might be useful to run an instance with >1 broker, e.g. `flux start --test-size=2 SCRIPT` and in the script ensure a 2 node "job" works.

**Reply by wspear** [reply]
Yep, that does the trick. I'd assumed the 'command' argument was for something flux specific, not just anything I want to throw at it. Thanks!
